source/tutorials/nixos-configuration-on-vm.md

```diff
@@ -0,0 +1,187 @@
+
+<!-- [old version 2022.08.10 should be deleted] -->
+<!-- The Git repository Nixpkgs provides the necessary infrastructure and information needed to specify a NixOS configuration[^nixpkgs]. -->
+<!-- Nixpkgs contains a library, packages, and modules. -->
+<!-- The library provides the functional infrastructure to conveniently create packages. -->
+<!-- The packages defined in Nixpkgs provide build scripts needed to build software. -->
+<!-- The NixOS modules are a way to configure the system. [previous formulation] provide a configuration infrastructure to generate the system's configuration files. -->
+
+## Starting from default NixOS Configuration
+
+On NixOS, the configuration file is normally located at `/etc/nixos/configuration.nix`.
+A different location can be specified using the environment variable `$NIX_PATH`.
+For CLI commands that use a configuration, the file path can be specified.
+In this guide this strategy is preferred because the goal is not to install or update NixOS, but to debug a configuration.
+
+On NixOS, you can use the `nixos-generate-config` command to create a configuration file that contains some useful defaults and configuration suggestions [^nixosconf].
+You can create a NixOS configuration in your working directory:
+```shell-session
+nixos-generate-config --dir ./
+```
+
+In the working directory you will find two files: `configuration.nix` and `hardware-configuration.nix`.
+The `hardware-configuration.nix` file is specific to your current installation.
+This file is irrelevant for this tutorial because it has no effect inside a virtual machine.
+The `configuration.nix` file contains various suggestions useful for the initial setup of a desktop computer.
+Without the comments the configuration file contains the following content:
+```nix
+{ config, pkgs, ... }:
+{
+  imports =
+    [
+      ./hardware-configuration.nix
+    ];
+
+  boot.loader.systemd-boot.enable = true;
+  boot.loader.efi.canTouchEfiVariables = true;
+
+  services.xserver.enable = true;
+
+  services.xserver.displayManager.gdm.enable = true;
+  services.xserver.desktopManager.gnome.enable = true;
+
+  system.stateVersion = "22.05";
+}
+```
+
+In this tutorial we focus on testing NixOS configurations on a virtual machine, in these cases you can remove the `hardware-configuration.nix` import.
+
+To be able to log in, you must uncomment the section that specifies the user "alice", or add the following lines to the configuration inside the curly bracket that contain the configuration specification:
+```nix
+  users.users.alice = {
+    isNormalUser = true;
+    extraGroups = [ "wheel" ];
+    packages = with pkgs; [
+      firefox
+      thunderbird
+    ];
+  };
+```
+
+Additionally, you need to specify a password for this user.
+An insecure solution is to specify an unencrypted plaintext password by adding the `initialPassword` option to the user configuration:
```

Comment:
I think unencrypted and plain text are the same thing.
```suggestion
An insecure solution is to specify an plain text password by adding the `initialPassword` option to the user configuration:
```

Reply:
There's no need for security in a test, so let's keep it simple.
This could be a nice example of configuration that you only add in the test and not in the production configuration.

```nix
nodes.machine = { ... }: {
  # ...
  users.users.alice.initialPassword = "test-password";
}
```

I'd also recommend `test-password`, because we don't need to tire the reader with unnecessary hints that something would be bad (boy who cried wolf).